### PR TITLE
chore: set up a `motoko` submodule

### PR DESCRIPTION
**Pull request by ggreif**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -155,7 +155,7 @@
     ],
     "scripts": {
         "vscode:prepublish": "run-s generate compile",
-        "generate:moc": "cd ../motoko && nix-shell --command 'make -C src moc.js && cp -vfL src/moc.js ../vscode-motoko/src/generated/moc.js'",
+        "generate:moc": "cd motoko && mv -vf .git .nogit && nix-shell --command 'mv -v .nogit .git && make -C src moc.js && cp -vfL src/moc.js ../src/generated/moc.js'",
```
Comment (ggreif) [inline package.json:158]: this is that `nix` assumes a fresh checkout (which it should be)

File: .vscodeignore [hunk 2]
```diff
@@ -1,3 +1,6 @@
 .github/**
+.gitmodules
 .gitignore
 .vscode/**
+motoko/**
+node_modules/**
```
Comment (ggreif) [inline .vscodeignore:6]: this brings down the `.vsix` size from 44MB to 7.3MB :-)

Comment (ggreif) [reply .vscodeignore:6]: we should do it on the `master` branch too!

Comment (ggreif) [reply .vscodeignore:6]: Are there any other files we could ignore for packaging?

**Review by rvanasa — APPROVED** [review]
LGTM; testing on my computer right now.

File: .vscodeignore (hunk 2 above)
Comment (rvanasa) [reply .vscodeignore:6]: Probably! I was thinking of using [Rollup](https://rollupjs.org/guide/en/) to tree-shake and bundle the TS code, but it seems like this also works.

File: README.md [hunk 3]
```diff
@@ -43,7 +43,9 @@ actor {
 
 ### Building from Source
 
-- Clone [motoko](https://github.com/dfinity/motoko) and [vscode-motoko](https://github.com/dfinity/vscode-motoko/tree/viper) into the same parent directory
-- Switch to the `viper` branch in both repositories
-- In your terminal, run `cd vscode-motoko` and then `npm run package` (this will rebuild the compiler bindings)
+- Clone [vscode-motoko](https://github.com/dfinity/vscode-motoko/tree/viper) into the same parent directory
```
Comment (ggreif) [inline README.md:46]:
```suggestion
- Clone [vscode-motoko](https://github.com/dfinity/vscode-motoko/tree/viper)
```

Comment (rvanasa) [reply README.md]: So much simpler now! 

Maybe we could specify `git clone https://github.com/dfinity/vscode-motoko --recurse-submodules` to set everything up in one swoop.

File: README.md [hunk 4]
```diff
@@ -43,7 +43,9 @@ actor {
...
 ### Building from Source
 
-- Clone [motoko](https://github.com/dfinity/motoko) and [vscode-motoko](https://github.com/dfinity/vscode-motoko/tree/viper) into the same parent directory
-- Switch to the `viper` branch in both repositories
-- In your terminal, run `cd vscode-motoko` and then `npm run package` (this will rebuild the compiler bindings)
+- Clone [vscode-motoko](https://github.com/dfinity/vscode-motoko/tree/viper)
+- In your terminal, run `cd vscode-motoko`
+- Switch to the `viper` branch (`git switch viper`)
+- Execute `git submodule update --init` to obtain the corresponding Motoko compiler
+- Run `npm run package` (this will rebuild the compiler bindings)
```
Comment (ggreif) [inline README.md:50]: `npm install` needed?

**Comment by rvanasa** [comment]
It seems like adding `node_modules` to the `.vscodeignore` file breaks the packaged version of the extension. We can get around this using a bundler such as [Rollup](https://rollupjs.org/guide/en/), which would also tree-shake unused code from the dependency tree.

**Comment by ggreif** [comment]
Yeah, I see the breakage. I saw some GH issue that the breakage can be due to the `devDependenies` containing stuff that should actually be in `dependencies`.

**Comment by rvanasa** [comment]
I'll try moving everything into `dependencies` in case that's the issue. My understanding is that we might need to use a bundler to get around this, but a simpler solution would be great.

**Comment by ggreif** [comment]
> I'll try moving everything into `dependencies` in case that's the issue

Nope, that sadly didn't work either :-(

**Comment by rvanasa** [comment]
Setting up a bundler right now, will ping you on the new PR.

File: README.md (hunk 3 above)
Comment (ggreif) [reply README.md]: good point let me try this flow...

Comment (ggreif) [reply README.md]: Okay, committed directly to `viper`. It worked!

**Comment by rvanasa** [comment]
#99

**Pull request merged**